aip/client-libraries/4235.md

```diff
@@ -0,0 +1,47 @@
+---
+id: 4235
+state: draft
+created: 2024-02-15
+---
+
+# Automatically populate fields in the request message
+
+For APIs that leverage request idempotency as described via [AIP-155], it could be useful to have the client libraries autogenerate the `request_id` if it is not already set by the customer.
+
+**Note:** This feature is primarily written for `request_id` fields within the request message. However, this feature should work for a field of any name, as long as the below conditions hold true.
+
+## Guidance
+
+APIs **may** configure fields in the request message for automatic population.
+If an API chooses to do so, they **must** guarantee idempotency for the configured fields.
+
+- If a duplicate request is detected, the server **should** return the
+  response for the previously successful request, because the client most
+  likely did not receive the previous response.
+- APIs **may** choose any reasonable timeframe for honoring request IDs.
+
+For a field to be automatically populated, the below configurations **must** be true:
+
+- The field **must** exist as a top-level string field of a unary method's request message (e.g. nested fields will not be eligible for automatic population).
+- The unary method which has the field in the request message **must** be explicitly designated in [google.api.client.Publishing.method_settings.selector] using the fully qualified name of the method.
+- The field **must** *not* be annotated with [google.api.field_behavior = REQUIRED].
+- The field name **must** be listed in [google.api.client.Publishing.method_settings.auto_populated_fields].
+- The field **must** be annotated with [google.api.field_info.format = UUID4].
+
+**Note:** This feature currently only supports automatic population of string UUID4 values. It may be expanded in the future to support other format types.
+
+### Generator configuration
```

Comment:
This isn't really generator configuration, the previous section was. This is more generator & client library _behavior_

Reply:
Updated the heading